xarray/tests/test_dataarray.py

```diff
@@ -1792,12 +1793,37 @@ def test_reindex_empty_array_dtype(self) -> None:
         x = xr.DataArray([], dims=("x",), coords={"x": []}).astype("float32")
         y = x.reindex(x=[1.0, 2.0])
 
-        assert x.dtype == y.dtype, (
-            "Dtype of reindexed DataArray should match dtype of the original DataArray"
-        )
-        assert y.dtype == np.float32, (
-            "Dtype of reindexed DataArray should remain float32"
-        )
+        assert (
+            x.dtype == y.dtype
+        ), "Dtype of reindexed DataArray should match dtype of the original DataArray"
+        assert (
+            y.dtype == np.float32
+        ), "Dtype of reindexed DataArray should remain float32"
+
+    def test_reindex_extension_array(self) -> None:
+        index1 = np.array([1, 2, 3])
+        index2 = np.array([1, 2, 4])
+        srs = pd.Series(index=index1, data=1).convert_dtypes()
+        x = srs.to_xarray()
+        y = x.reindex(index=index2)  # used to fail (GH #10301)
+        assert_array_equal(x, pd.array([1, 1, 1]))
+        assert_array_equal(y, pd.array([1, 1, pd.NA]))
+        assert x.dtype == y.dtype == pd.Int64Dtype()
+        assert x.index.dtype == y.index.dtype == np.dtype("int64")
+
+    def test_reindex_categorical(self) -> None:
+        index1 = pd.Categorical(["a", "b", "c"])
```

Comment:
I believe this test was going through a different codepath than the one intended, actually.  Categorical indices are handled differenlty if I'm not mistaken